conans/test/functional/compression/compression_test.py

```diff
@@ -0,0 +1,454 @@
+
+class TarGzCompressor(TarCompressor):
+    read_mode = "r:gz"
+    write_mode = "w:gz"
+    ext = ".tgz"
+
+
+class TarBz2Compressor(TarCompressor):
+    read_mode = "r:bz2"
+    write_mode = "w:bz2"
+    ext = ".tbz2"
+
+
+class TarLZMACompressor(TarCompressor):
+    read_mode = "r:xz"
+    write_mode = "w:xz"
+    ext = ".txz"
+
+
+compressors = [{"compressor": ZipCompressor()},
+               {"compressor": TarCompressor()},
+               {"compressor": TarGzCompressor()},
+               {"compressor": TarBz2Compressor()}]
+
+if sys.version_info.major >= 3:
+    compressors.append({"compressor": TarLZMACompressor()})
+
+
+# check if we can actually support symlinks:
+# some operation systems might not support them
+# e.g. Windows 10 requires either privileges or developer mode
+# also only certain file systems have support (e.g. NTFS does, but FAT32 doesn't)
+def symlinks_supported():
+    if not hasattr(os, "symlink"):
+        return False
+    client = TestClient()
+    try:
+        client.save({"a": ""})
+        with chdir(client.current_folder):
+            os.symlink("a", "b")
+            return os.path.islink("b")
+    except OSError:
+        return False
+
+
+def hardlinks_supported():
+    if not hasattr(os, "link"):
+        return False
+    client = TestClient()
+    try:
+        client.save({"a": ""})
+        with chdir(client.current_folder):
+            os.link("a", "b")
+            return os.stat("a").st_ino == os.stat("b").st_ino
+    except OSError:
+        return False
+
+
+@parameterized_class(compressors)
+@attr("slow")
+class CompressionTest(unittest.TestCase):
+    @classmethod
+    def get_file_list(cls, d):
+        files = []
+        dirs = []
+        links = []
+        for root, dirnames, filenames in os.walk(d):
+            for filename in filenames:
+                relpath = os.path.relpath(os.path.join(root, filename), d)
+                if os.path.islink(os.path.join(root, filename)):
+                    links.append(relpath)
+                else:
+                    files.append(relpath)
+            for dirname in dirnames:
+                relpath = os.path.relpath(os.path.join(root, dirname), d)
+                if os.path.islink(os.path.join(root, dirname)):
+                    links.append(relpath)
+                else:
+                    dirs.append(relpath)
+        return files, dirs, links
+
+    @classmethod
+    def is_broken_link(cls, link):
+        if os.path.islink(link):
+            target = os.readlink(link)
+            return not os.path.exists(target)
+        return False
+
+    @classmethod
+    def get_inodes(cls, d, files):
+        inodes = dict()
+        for filename in files:
+            stat = os.stat(os.path.join(d, filename))
+            if stat.st_ino not in inodes:
+                inodes[stat.st_ino] = []
+            inodes[stat.st_ino].append(filename)
+        return sorted(list(inodes.values()))
+
+    def compare_dirs(self, d1, d2):
```

Comment:
Call it ``assertDirsEqual``, or ``assert_folders_match()`` if this will be raising test assertions

Reply:
renamed to assertDirsEqual